cylc/flow/task_events_mgr.py

```diff
@@ -1277,6 +1278,7 @@ def _insert_task_job(
                 'submit_status': submit_status,
                 'time_submit_exit': event_time,
                 'job_id': itask.summary.get('submit_method_id'),
+                'flow_nums': serialise(itask.flow_nums),
```

Comment:
I'm not sure we need to set this here. I think `put_update_task_jobs` should be called only with fields which have changed. The `flow_nums` field should be set when the job is added to the DB but never changed.

I think it's the `put_insert_task_jobs` calls which need this information.